chipsec/logger.py

```diff
@@ -51,14 +51,77 @@
 
 LOGGER_NAME = 'CHIPSEC'
 
-class chipsecRecordFactory(logging.LogRecord):
+
+class level(Enum):
+    DEBUG = 10
+    HELPER = 11
+    HAL = 12
+    VERBOSE = 13
+    INFO = 20
+    GOOD = 21
+    BAD = 22
+    IMPORTANT = 23
+    WARNING = 30
+    ERROR = 40
+    CRITICAL = 50
+    EXCEPTION = 60
+
+
+class chipsecFilter(logging.Filter):
+    def __init__(self, name: str = ...) -> None:
+        super().__init__(name)
+
+    def filter(self, record):
+        if record.levelno == level.ERROR.value:
+            record.additional = 'ERROR: '
+        elif record.levelno == level.WARNING.value:
+            record.additional = 'WARNING: '
+        elif record.levelno == level.IMPORTANT.value:
+            record.additional = '[!] '
+        elif record.levelno == level.GOOD.value:
+            record.additional = '[+] '
+        elif record.levelno == level.BAD.value:
+            record.additional = '[-] '
+        elif record.levelno == level.DEBUG.value:
+            record.additional = '[*] [DEBUG] '
+        elif record.levelno == level.VERBOSE.value:
+            record.additional = '[*] [VERBOSE] '
+        elif record.levelno == level.HAL.value:
+            record.additional = '[*] [HAL] '
+        elif record.levelno == level.HELPER.value:
+            record.additional = '[*] [HELPER] '
+        else:
+            record.additional = ''
+        return True
+
+
+class chipsecLogFormatter(logging.Formatter):
+    def __init__(self, fmt: Optional[T] = ..., datefmt: Optional[T] = ..., style='%') -> None:
```

Comment:
```suggestion
    def __init__(self, fmt: Optional[str] = ..., datefmt: Optional[str] = ..., style='%') -> None:
```
Sorry, T was just a placeholder. Please keep the correct type. Everything else is looking good!